spring-xsuaa/src/main/java/com/sap/cloud/security/xsuaa/token/authentication/ReactiveXsuaaJwtDecoder.java

```diff
@@ -0,0 +1,87 @@
+package com.sap.cloud.security.xsuaa.token.authentication;
+
+import java.text.ParseException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import org.springframework.security.oauth2.core.DelegatingOAuth2TokenValidator;
+import org.springframework.security.oauth2.core.OAuth2TokenValidator;
+import org.springframework.security.oauth2.jwt.Jwt;
+import org.springframework.security.oauth2.jwt.JwtException;
+import org.springframework.security.oauth2.jwt.JwtTimestampValidator;
+import org.springframework.security.oauth2.jwt.NimbusReactiveJwtDecoder;
+import org.springframework.security.oauth2.jwt.ReactiveJwtDecoder;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.nimbusds.jwt.JWT;
+import com.nimbusds.jwt.JWTParser;
+import com.sap.cloud.security.xsuaa.XsuaaServiceConfiguration;
+
+import net.minidev.json.JSONObject;
+import reactor.core.publisher.Mono;
+
+public class ReactiveXsuaaJwtDecoder implements ReactiveJwtDecoder {
+
+	Cache<String, ReactiveJwtDecoder> cache;
+	private final XsuaaServiceConfiguration xsuaaServiceConfiguration;
+	private List<OAuth2TokenValidator<Jwt>> tokenValidators = new ArrayList<>();
+
+	private static final String EXT_ATTR = "ext_attr";
+	private static final String ZDN = "zdn";
+	private static final String ZID = "zid";
+
+	// var arg it is only being converted to a List<OAuth2TokenValidator<Jwt>>,
+	// therefore its type safe.
+	@SafeVarargs
+	ReactiveXsuaaJwtDecoder(XsuaaServiceConfiguration xsuaaServiceConfiguration, int cacheValidity, int cacheSize,
+			OAuth2TokenValidator<Jwt>... tokenValidators) {
```

Comment:
The constructor is package-private so I assume intended use is only via `XsuaaJwtDecoderBuilder`. Then I'd suggest changing the parameter `tokenValidators` to the concrete type instead of vararg.

Reply:
Done -> Thanks!